### ISSUE TITLE
`union` method returns `NoMethodError` in Rails 8

### ISSUE THREAD

**Issue by JuzerShakir** [brianhempel/active_record_union#38]
I have 2 scopes:

1.   `scope :no_thaali, -> { where.missing(:thaalis) }`
2.   `scope :took_thaali, -> { thaalis.group("sabeels.id").having("MAX(thaalis.year) < #{CURR_YR}") }`

I want another scope that is the concatenation of the results of these 2 scopes.

In Rails 7.1 & 7.2, this was working perfectly fine:
```
scope :not_taking_thaali, -> { no_thaali.union(took_thaali) }
```

After upgrading to Rails 8 and running test suites, it returns this error message:
```
Failure/Error: scope :not_taking_thaali, -> { no_thaali.union(took_thaali) }
     
     NoMethodError:
       undefined method `arel_table' for nil
     # /usr/share/rvm/gems/ruby-3.3.6/gems/active_record_union-1.3.0/lib/active_record_union/active_record/relation/union.rb:40:in `set_operation'
     # /usr/share/rvm/gems/ruby-3.3.6/gems/active_record_union-1.3.0/lib/active_record_union/active_record/relation/union.rb:11:in `union'
```

So does this gem support Rails 8, yet? If yes, then what am I missing here? Is something changed that I am not aware of? Thank you guys! :)

**Comment by JuzerShakir** [comment]
I investigated the codebase and found that you've already supported Rails 8. 
 
 To debug the issue locally, I opened the `active_record_union` gem using `bundle open`. However, the code in the local version wasn't up-to-date with the remote repo on the `master` branch. 
 
 To ensure I was working with the latest code, I updated my Gemfile to fetch the `main` branch directly from GitHub. This resolved the issue and all tests passed.

**Comment by JuzerShakir** [comment]
I investigated the codebase and found that you've already supported Rails 8. 
 
 To debug the issue locally, I opened the `active_record_union` gem using `bundle open`. However, the code in the local version wasn't up-to-date with the remote repo on the `master` branch. 
 
 To ensure I was working with the latest code, I updated my Gemfile to fetch the `main` branch directly from GitHub. This resolved the issue and all tests passed.
